addons/mail/models/mail_message.py

```diff
@@ -1034,6 +1033,68 @@ def message_format(self, format_reply=True, msg_vals=None):
                 vals['module_icon'] = modules.module.get_module_icon(self.env[vals['model']]._original_module)
         return vals_list
 
+    @api.model
+    def _message_format_personalized_prepare(self, messages_formatted, partner_ids=None):
+        """ Prepare message to be personalized by partner.
+
+        This method add partner information in batch to the messages so that the
+        messages could be personalized for each partner by using
+        _message_format_personalize with no or a limited number of queries.
+
+        For example, it gathers all followers of the record related to the messages
+        in one go (or limit it to the partner given in parameter), so that the method
+        _message_format_personalize could then personalize each message for each partner.
+
+        :param list messages_formatted: list of message formatted using the method
+            message_format
+        :param list partner_ids: (optional) limit value computation to the partners
+            of the given list; if not set, all partners are considered (all partners
+            following each record will be included in follower_id_by_partner_id).
+
+        :return: list of messages_formatted with added value:
+            'follower_id_by_partner_id': dict partner_id -> follower_id of the record
+        """
+        domain = expression.OR([
+            [('res_model', '=', model), ('res_id', 'in', list({value['res_id'] for value in values}))]
+            for model, values in tools_groupby(
+                (vals for vals in messages_formatted if vals.get("model") and vals.get("res_id")),
+                key=lambda r: r["model"]
+            )
+        ])
+        if partner_ids:
+            domain = expression.AND([domain, [('partner_id', 'in', partner_ids)]])
+        records_followed = self.env['mail.followers'].sudo().search(domain)
```

Comment:
@seb-odoo indeed, we could probably do a small hack to explicitely remove channel records, as it will only be until we split models so probably not so long cc @pyduf 

By the way: '_get_mail_recipients_follower_status' should also be skipped for channels while we are at it, as it is quite symmetric

Reply:
I have added a condition in both method to skip message related to dicuss.channel model. Is it what you had in mind or did I miss something ?